src/lib/keycard-pdf.ts

```diff
@@ -0,0 +1,227 @@
+import PDFDocument from 'pdfkit'
+import { Base64Encode } from 'base64-stream'
+import * as QRCode from 'qrcode'
+
+const QR_DATA_GAP = 15
+const QR_WIDTH = 128
+const DEFAULT_LOGO_PATH = './resources/sml-logo.png'
+
+const colors = {
+  REGULAR_COLOR: '#000',
+  TEXT_COLOR: '#3a3a3a',
+  INFO_COLOR: '#7a7a7a',
+  WARN_BG_COLOR: '#ffe4dd',
+  WARN_TEXT_COLOR: '#ff1c23'
+}
+
+const fonts = {
+  MONOSPACE_FONT: 'Monospace Font',
+  REGULAR_FONT: 'Regular Font',
+  STRONG_FONT: 'Strong Font'
+}
+
+const fontSizes = {
+  DOCUMENT_TITLE: 24,
+  HEADER_1: 20,
+  HEADER_3: 14,
+  HEADER_4: 12,
+  REGULAR: 10,
+  SMALL: 8
+}
+
+export const generatePdf = async (walletName: string, userPrivateKey: string, backupPrivateKey: string, servicePublicKey: string, logoPath?: string): Promise<string> => {
+  const doc = await generateBackupPdf(walletName, userPrivateKey, backupPrivateKey, servicePublicKey, logoPath)
+  const stream = doc.pipe(new Base64Encode())
+  let base64Pdf: string = ''
+  stream.on('data', (data: string) => base64Pdf += data)
+  return new Promise<string>((resolve: Function, reject: Function) => {
+    stream.on('end', () => resolve(base64Pdf))
+    stream.on('error', () => reject())
+  })
+}
+
+const generateBackupPdf = async (walletName: string, userPrivateKey: string, backupPrivateKey: string, servicePublicKey: string, logoPath?: string): Promise<PDFKit.PDFDocument> => {
+  const doc = new PDFDocument()
+  doc.registerFont(fonts.MONOSPACE_FONT, 'Courier')
+  doc.registerFont(fonts.REGULAR_FONT, 'Helvetica')
+  doc.registerFont(fonts.STRONG_FONT, 'Helvetica-Bold')
+  const start = 140
+  const imageHeight = 15
+  const qrOffset = 50
+
+  addLogo(doc, logoPath || DEFAULT_LOGO_PATH, imageHeight)
+  addGeneratedAt(doc, walletName)
+  printFlash(doc, 'Print this page or keep it securely offline. See second page for FAQ', 0, 70)
+
+  await drawDataBox(doc, {
+    header: 'A: User Key',
+    description: 'This is your password-encrypted private key',
+    data: userPrivateKey,
+    x: doc.page.margins.left,
+    y: (start + qrOffset)
+  })
+
+  await drawDataBox(doc, {
+    header: 'B: Backup Key',
+    description: 'This is your password-encrypted backup private key',
+    data: backupPrivateKey,
+    x: doc.page.margins.left,
+    y: 2 * (start + qrOffset)
+  })
+
+  await drawDataBox(doc, {
+    header: 'C: Service Public Key',
+    description: 'This is the public part of the key the service will use to co-sign transactions with you on Your wallet',
+    data: servicePublicKey,
+    x: doc.page.margins.left,
+    y: 3 * (start + qrOffset)
+  })
+
+  // await drawDataBox(doc, {
+  //   header: 'D: Encrypted wallet password',
+  //   description: 'This is your password-encrypted private key',
+  //   data: encryptedPassword,
+  //   x: doc.page.margins.left,
+  //   y: start * 4 + 60
+  // })
```

Comment:
A co z tym?

Reply:
To jest do dorobienia w osobnym tasku, bo wymaga jeszcze zmian w sakiewce-api